File: python_backend/game_logic/ranges/get_ranges_dict.py
from pprint import pprint
from game_logic.test_objects.sample_board_dicts import sample_board_dicts


def get_range_dict(board):
    """initialize dict with key = piece id and value = range of piece"""
    ranges = {}
    for id_ in board.values():
        if id_ == '#':
            continue
        else:
            ranges[id_] = []

    return ranges


if __name__ == "__main__":
    board = sample_board_dicts['ranges_test']
    ranges = get_range_dict(board)
    pprint(ranges)

File: node.js/ranges/getRangesDict.js
var  pprint  = require("pprint");
var  sampleBoardDicts  = require("gameLogic/testObjects/sampleBoardDicts");
const getRangeFunc = require("./getRangeFunc");


function getRangeDict(board) {
    /*initialize dict with let key = piece id &&  let value = range of piece**/
    let ranges = {}
    for (var id_ of Object.values(board)) {
        if (id_ === '#') {
            continue
        }
        else {
            ranges[id_] = []
        }
    }
    return ranges
}

module.exports = getRangeDict;

if (require.main === module) {
    let board = sampleBoardDicts['ranges_test']
    ranges = getRangeDict(board)
    console.log(ranges)
}
